### PR TITLE
Handle `!ConvertEmptyStringToNull` cases correctly in `SimpleTypeModelBinder`

### PR DESCRIPTION
**Pull request by dougbu**
- #4988
- preserve whitespace as the setting demands
 - correct previous `string.IsNullOrEmpty()` call to match previous `ValueProviderResultExtensions.ConvertTo()` use
- short-circuit other `string`-to-`string` conversions (as `ValueProviderResultExtensions.ConvertTo()` does)
- correct documentation of `ConvertEmptyStringToNull` properties
- add more tests of these scenarios and remove duplicate `BindModel_ValidValueProviderResult_ConvertEmptyStringsToNull()` test

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/SimpleTypeModelBinder.cs [hunk 1]
```diff
@@ -47,23 +47,30 @@ public Task BindModelAsync(ModelBindingContext bindingContext)
...
-                }
-
+                object model;
                 if (bindingContext.ModelType == typeof(string))
                 {
-                    var modelAsString = model as string;
-                    if (bindingContext.ModelMetadata.ConvertEmptyStringToNull &&
-                        string.IsNullOrEmpty(modelAsString))
+                    // Already have a string. No further conversion required but handle ConvertEmptyStringToNull.
+                    if (bindingContext.ModelMetadata.ConvertEmptyStringToNull && string.IsNullOrWhiteSpace(value))
```
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/SimpleTypeModelBinder.cs]: MVC 5 and ASP.NET 3 both use `string.IsNullOrWhiteSpace()` in their analogous code. Same for `ValueProviderResultExtensions.ConvertTo()` which this code used prior to 4e97c72. (It's that commit that broke `!ConvertEmptyStringToNull` and added useless type conversions from `string` to `string`.)

**Comment by dougbu** [comment]
thanx for volunteering @pranavkm!

File: src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/SimpleTypeModelBinder.cs [hunk 2]
```diff
@@ -47,23 +47,30 @@ public Task BindModelAsync(ModelBindingContext bindingContext)
...
                         model = null;
                     }
+                    else
+                    {
+                        model = value;
+                    }
+                }
+                else if (string.IsNullOrWhiteSpace(value))
+                {
+                    // Other than the StringConverter, converters Trim() the value then throw if the result is empty.
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/SimpleTypeModelBinder.cs]: We specifically care about converting non-empty strings so documenting the behavior of type converters for this scenario doesn't help.

File: src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/SimpleTypeModelBinder.cs (hunk 1 above)
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/SimpleTypeModelBinder.cs]: Would `_typeConverter.ConvertFrom` trimmed the string in the past?

Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/SimpleTypeModelBinder.cs]: No, the `StringConverter` just returns a `string` source unchanged. It's about the only `TypeConverter` that does _not_ `Trim()`.

Note I did not have to change any existing tests except to remove a duplicate.

File: src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/SimpleTypeModelBinder.cs (hunk 2 above)
Comment (dougbu) [inline src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/SimpleTypeModelBinder.cs]: :see_no_evil: Agree there'd be no reason to document anything here if our reasoning stopped at "don't call type converters w/ whitespace (or null)". But it's a bit deeper than that.

**Comment by pranavkm** [comment]
:shipit:

**Comment by dougbu** [comment]
fae0e9a
